pycalphad/io/database.py

```diff
@@ -210,6 +211,7 @@ def from_file(fname, fmt=None):
         try:
             dbf = Database()
             format_registry[fmt.lower()].read(dbf, fd)
+            dbf._parameter_queue_process()
```

Comment:
Would it make more sense for the format parsers to call `dbf._parameter_queue_process()`, since they would be the ones override `force_insert`? Trying to think about ways that the database could possibly be put into an inconsistent state by adding parameters manually.

Alternatively, would it make sense to come up with a solution to update process the queue automatically, maybe when when accessing `_parameters`?

Reply:
1. Yes, I'll move the function call.
2. It should be difficult to get the database into an inconsistent state: `force_insert` is designed to use the old behavior by default, so code unaware of this change should work the same.
3. Some kind of lazy/dynamic parsing of parameters would be a UX improvement here, definitely, since in most cases the vast majority of parameters are unused in downstream calculations. Getting this right would probably require breaking the API used by Model, or some really clever thinking.

Reply:
A hack could just be to change the tinydb object (`self._parameters`) to `self._parameter_database = TinyDB(storage=MemoryStorage)` probably needs a better name)

```python
@property
def _parameters():
    if len(self._parameter_queue()) > 0:
        self.process_parameter_queue()
    return self._parameter_database
```

Reply:
Oh, I see. We're talking about two different things. That's a good idea.